### PR TITLE
feat(framework): scope the createMock expectations rule via phpstan configuration

### PR DESCRIPTION
**Pull request by nfortier-shopware**
### 1. Why is this change necessary?

NoCreateMockWithoutExpectationsRule hardcodes the enforced test namespaces, so downstream consumers cannot roll the rule out over their own test suites domain by domain.

### 2. What does this change do, exactly?

- The rule reads the enforced namespaces from a new `shopware.createMockWithoutExpectationsEnabledNamespaces` PHPStan parameter (default: `Shopware\Tests\Unit\`, so core enforcement is unchanged) instead of the hardcoded list. An empty list disables the rule.
- The parameter is the sole scope control: the rule gates on `TestRuleHelper::isTestClass()` plus the enabled namespaces, independent of `allowedUnitTestClassNamespaces`. Any consumer can opt in any of its test suites (unit, migration, integration, other plugins) by listing the namespaces, and consumers that keep the default are unaffected.
- The devops rule tests cover the new wiring: a narrowing test proves an enabled-namespaces list silences classes outside it, and a disabled test proves the empty list turns the rule off.

### 3. Describe each step to reproduce the issue or behaviour.

Run PHPStan in a consumer whose unit tests are not under `Shopware\Tests\Unit\`: the rule never fires there, and there is no configuration to enable it for a chosen namespace.

### 4. Please link to the relevant issues (if any).

- relates #17937

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have updated developer-facing release notes if this change is **relevant** for external developers:
  - Add a short entry to `RELEASE_INFO-6.<major>.md` under “Upcoming” for informational changes, including the consequences of the change and how it affects external developers.
  - Add an `UPGRADE` section in `UPGRADE-6.<next-major>.md` for breaking changes (what/why/impact/how to adapt).
  - See the [Documenting a Release Process](https://github.com/shopware/shopware/blob/trunk/delivery-process/documenting-a-release.md) for details.
- [ ] I have written or adjusted the documentation and agent skills according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfilled them

### CODE REVIEW

File: src/Core/DevOps/StaticAnalyze/PHPStan/Rules/Tests/NoCreateMockWithoutExpectationsRule.php [hunk 1]
```diff
@@ -84,8 +73,37 @@ class NoCreateMockWithoutExpectationsRule implements Rule
...
 
-    public function __construct(private readonly Parser $parser)
-    {
+    /**
+     * Hard boundary: the rule never fires outside these trees, so third-party plugins reusing the
+     * core PHPStan config do not inherit enforcement. Consumers narrow further via $enabledNamespaces.
+     *
+     * @var list<string>
+     */
+    private const SUPPORTED_NAMESPACES = [
```
Comment (keulinho) [inline src/Core/DevOps/StaticAnalyze/PHPStan/Rules/Tests/NoCreateMockWithoutExpectationsRule.php]: why do we still need this when we moved it to config? When someone wants to use the rule by actively configuring it, why do we block that?

i'm also mainly thinking about our own plugins, e.g is there a reason we can't opt in for this rule in the paypal plugin? Because there we need to prepare for the phpunit upgrade as well? 🤔

Comment (nfortier-shopware) [reply src/Core/DevOps/StaticAnalyze/PHPStan/Rules/Tests/NoCreateMockWithoutExpectationsRule.php]: Maybe it's too extra careful approach indeed, there is no reason we can't. Will lift it 👍

File: src/Core/DevOps/StaticAnalyze/PHPStan/Rules/Tests/NoCreateMockWithoutExpectationsRule.php [hunk 2]
```diff
@@ -1291,7 +1309,19 @@ private function buildError(Node $createMockCall, int $line, string $message, ?s
 
     private function isEnabledNamespace(string $className): bool
     {
-        foreach (self::ENABLED_NAMESPACES as $namespace) {
+        if (!self::matchesAny($className, self::SUPPORTED_NAMESPACES)) {
+            return false;
+        }
+
+        return $this->enabledNamespaces === [] || self::matchesAny($className, $this->enabledNamespaces);
```
Comment (umutdogan4291) [inline src/Core/DevOps/StaticAnalyze/PHPStan/Rules/Tests/NoCreateMockWithoutExpectationsRule.php]: if `enabledNamespaces` is empty `isEnabledNamespace` must return false 🤔 now returns true

Comment (nfortier-shopware) [reply src/Core/DevOps/StaticAnalyze/PHPStan/Rules/Tests/NoCreateMockWithoutExpectationsRule.php]: this was illogical indeed, will fix

**Review by umutdogan4291 — APPROVED** [review]

File: src/Core/DevOps/StaticAnalyze/PHPStan/Rules/Tests/NoCreateMockWithoutExpectationsRule.php (hunk 1 above)
Comment (nfortier-shopware) [reply src/Core/DevOps/StaticAnalyze/PHPStan/Rules/Tests/NoCreateMockWithoutExpectationsRule.php]: Done, it gives idea of follow up for many other rules we have, there are too many ways to gate them right now

**Review by keulinho — APPROVED** [review]
